src/uct/ib/dc/dc_mlx5_ep.c

```diff
@@ -120,9 +119,19 @@ uct_dc_mlx5_ep_atomic_op_post(uct_ep_h tl_ep, unsigned opcode, unsigned size,
         return status;
     }
 
-    UCT_RC_IFACE_GET_TX_ATOMIC_DESC(&iface->super.super, &iface->super.tx.atomic_desc_mp, desc);
+    UCT_DC_MLX5_CHECK_RES(iface, ep);
```

Comment:
we need to avoid doing something noticeable before checking resources. Otherwise it may affect performance
Please check this as a veri first thing in this func.
(then can use `goto` if  `uct_rc_mlx5_iface_common_atomic_data` fails)